platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/tipset/Tipset.java

```diff
@@ -0,0 +1,254 @@
+     */
+    public long getWeightedAdvancementCount(final long nodeId, @NonNull final Tipset that) {
+        long count = 0;
+
+        final int selfIndex = nodeIdToIndex.applyAsInt(nodeId);
+        for (int index = 0; index < tips.length; index++) {
+            if (index == selfIndex) {
+                // We don't consider self advancement here, since self advancement does nothing to help consensus.
+                continue;
+            }
+
+            if (this.tips[index] < that.tips[index]) {
+                count += indexToWeight.applyAsLong(index);
+            }
+        }
+
+        return count;
+    }
+
+    /**
+     * <p>
+     * Get the number of tip advancements, weighted using the provided function, between this tipset and another
+     * tipset.
+     * </p>
+     *
+     * <p>
+     * A tip advancement is defined as an increase in the tip generation for a node ID. The exception to this rule is
+     * that an increase in generation for the target node ID is never counted as a tip advancement. The tip advancement
+     * count is defined as the sum of all tip advancements after being appropriately weighted.
+     * </p>
+     *
+     * @param nodeId compute the advancement count relative to this node ID
+     * @param that   the tipset to compare to
+     * @return the number of tip advancements to get from this tipset to that tipset
+     */
+    public long getWeightedAdvancementCount(
+            final long nodeId, @NonNull final Tipset that, @NonNull IntToLongFunction indexWeight) {
+        long count = 0;
+
+        final int selfIndex = nodeIdToIndex.applyAsInt(nodeId);
+        for (int index = 0; index < tips.length; index++) {
+            if (index == selfIndex) {
+                // We don't consider self advancement here, since self advancement does nothing to help consensus.
+                continue;
+            }
+
+            if (this.tips[index] < that.tips[index]) {
+                count += indexWeight.applyAsLong(index);
+            }
+        }
+
+        return count;
+    }
+
+    // TODO javadoc
+    public int getAdvancementCount(final long nodeId, @NonNull final Tipset that) {
+        int count = 0;
+
+        final int selfIndex = nodeIdToIndex.applyAsInt(nodeId);
+        for (int index = 0; index < tips.length; index++) {
+            if (index == selfIndex) {
+                // We don't consider self advancement here, since self advancement does nothing to help consensus.
+                continue;
+            }
+
+            if (this.tips[index] < that.tips[index]) {
+                count += 1;
+            }
+        }
+
+        return count;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        final StringBuilder sb = new StringBuilder("(");
+        for (int index = 0; index < tips.length; index++) {
+            sb.append(index).append(":").append(tips[index]);
```

Comment:
Here's another microoptimization that buys you nothing except, arguably, easier to read code:  Always append `", "` in the loop.  Then on exiting the loop: `if (sb.length() > 1) sb.setLength(sb.length()-2);`.  Anyway, _I'd_ argue it's easier to read, YMMV.